### ISSUE TITLE
Installing man pages globally

### ISSUE THREAD

**Issue by TheKhanj** [volta-cli/volta#1708]
I was looking for [installing man pages](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#man) using volta, I couldn't find any, neither in github issues nor anywhere else. Is this feature implemented in volta? If not is there any plan for it to be implemented? It might be my fault not being able to find it, if that's the case pardon me for opening an issue for it 🙏

Here's the [link](https://github.com/npm/cli/blob/latest/node_modules/bin-links/lib/link-mans.js) to npm's implementation for it.

**Comment by chriskrycho** [comment]
Thanks for opening this, and you did not miss anything. We do not currently supply man pages! However, it would be *relatively* straightforward to implement using [the `clap_mangen` crate](https://crates.io/crates/clap_mangen), since we already use clap for our CLI parser. If someone wants to implement it, it should be a *relatively* easy first issue.

For a rough idea of how to implement it, you would want to do roughly [what we do in `src/command/completions.rs`](https://github.com/volta-cli/volta/blob/main/src/command/completions.rs):

- Add `clap_mangen` as a dependency for that crate.
- Add a new `man_pages` module as a sibling to `completions` and the others.
- Define a new clap argument with a `pub(crate) struct ManPages`.
- Implement Volta’s `Command` trait for `ManPages`, including logging events the same basic way `Completionts` does (so you will also need to add a new `ActivityKind` etc.) for it.
- Use the `Man` struct from `clap_mangen` to generate the man pages from the command (`crate::cli::Volta::command()`).
- Emit the generated roff to the appropriate location. (You will have to see where that is on a per-OS basis!) One note that adds a *small* bit of complexity here: there is no “default” place to install it on Windows, as far as I know, but we should presumably still support it for someone who wants to read them there.

**Comment by TheKhanj** [comment]
Well I would like to work on that myself, it should be fun :).

**Comment by chriskrycho** [comment]
Ahhhhh, I see. That is a *separate* very good feature request, and I should have clicked the links—I thought you were talking about npm’s implementation *for itself* rather than *for packages*. Whoops! We don’t have the feature explicitly… but you should test it? Because under the hood, `volta install` is *using* `npm install`, `yarn install`, etc., so if it works in the underlying package manager, it may (I will not say “should” here, but “may”!) work already. Perhaps you’re asking, though, because you see that it does not?

**Comment by TheKhanj** [comment]
Yeah, that didn't work for me. I should really check volta's source code to be sure I haven't missed anything though. But the way npm handles it is that it only symlinks man pages when the package is installed **globally** and the system is unix.

```typescript
const binLinks = opts => {
  const { path, pkg, force, global, top } = opts
  // global top pkgs on windows get bins installed in {prefix}, and no mans
  //
  // unix global top pkgs get their bins installed in {prefix}/bin,
  // and mans in {prefix}/share/man
  //
  // non-top pkgs get their bins installed in {prefix}/node_modules/.bin,
  // and do not install mans
  //
  // non-global top pkgs don't have any bins or mans linked.  From here on
  // out, if it's top, we know that it's global, so no need to pass that
  // option further down the stack.
  if (top && !global) {
    return Promise.resolve()
  }

  return Promise.all([
    // allow clobbering within the local node_modules/.bin folder.
    // only global bins are protected in this way, or else it is
    // yet another vector for excessive dependency conflicts.
    linkBins({ path, pkg, top, force: force || !top }),
    linkMans({ path, pkg, top, force }),
  ])
}

```

**Comment by chriskrycho** [comment]
Intuitively, it is *likely* something about the way it does the final link actions—IIRC we *do* run `npm install --global` but provide it a specific installation path to treat as its global installation location—but you’d have to set up some kind of debugging in the Node-side code to confirm that. That might be a touch annoying, since you will need to make sure you have the relevant copy of Node on your system which the `volta install` will go through, and be *sure* it’s going through that one, and then patch the version in Volta’s cache to add logging or whatever. 😬
